service/share/light_availability.go

```diff
@@ -0,0 +1,75 @@
+package share
+
+import (
+	"context"
+	"errors"
+
+	format "github.com/ipfs/go-ipld-format"
+
+	"github.com/celestiaorg/celestia-node/ipld"
+)
+
+// DefaultSampleAmount sets the default amount of samples to be sampled from the network by lightAvailability.
+var DefaultSampleAmount = 16
+
+// lightAvailability implements Availability using Data Availability Sampling technique.
+// It is light because it does not require to download all the data to verify it availability.
+// It is assumed that there are lots of lightAvailability instances on the network doing sampling over the same Root
+// to collectively verify its availability.
+type lightAvailability struct {
+	getter format.NodeGetter
+}
+
+// NewLightAvailability creates a new Light DataAvailability.
+func NewLightAvailability(get format.NodeGetter) Availability {
+	return &lightAvailability{
+		getter: get,
+	}
+}
+
+// SharesAvailable randomly samples DefaultSamples amount of Shares committed to the given Root.
+// This way SharesAvailable subjectively verifies that Shares are available.
+func (la *lightAvailability) SharesAvailable(ctx context.Context, dah *Root) error {
+	log.Debugw("Validate availability", "root", dah.Hash())
+	samples, err := SampleSquare(len(dah.ColumnRoots), DefaultSampleAmount)
```

Comment:
Aside: can there be a common size like `dah.size`? It seems a bit hacky to use the length of column roots but not row roots. A reader might think "why?"

More to the point, can you make a TODO for the number of samples here?

Reply:
I think this is the right point! I will change to RowRoots for consistency. Adding Size method to DAH is also a good option but that will require a PR to core where DAH is defined and another loop of reviews we don't have a time for.

Reply:
Can you just make an issue so we don't forget eventually?

Reply:
@adlerjohn, https://github.com/celestiaorg/celestia-core/issues/575